stl/inc/ranges

```diff
@@ -6837,6 +6837,664 @@ namespace ranges {
         inline constexpr _Stride_fn stride;
     } // namespace views
 
+#ifdef __cpp_lib_ranges_zip
+    namespace _Zip {
+        template <typename... _RangeTypes>
+        concept _Is_common = (sizeof...(_RangeTypes) == 1 && (_RANGES common_range<_RangeTypes> && ...)) //
+                          || (!(_RANGES bidirectional_range<_RangeTypes> && ...)
+                              && (_RANGES common_range<_RangeTypes> && ...)) //
+                          || ((_RANGES random_access_range<_RangeTypes> && ...)
+                              && (_RANGES sized_range<_RangeTypes> && ...));
+
+#ifndef __cpp_lib_tuple_like
+        template <typename... _RangeTypes>
+        struct _Tuple_or_pair_solver {
+            using _Storage_type = _STD tuple<_RangeTypes...>;
+        };
+
+        template <typename... _RangeTypes>
+            requires(sizeof...(_RangeTypes) == 2)
+        struct _Tuple_or_pair_solver<_RangeTypes...> {
+            using _Storage_type = _STD pair<_RangeTypes...>;
+        };
+
+        template <typename... _RangeTypes>
+        using _Tuple_or_pair = typename _Tuple_or_pair_solver<_RangeTypes...>::_Storage_type;
+#else // ^ Pre-P2165R4 / Post-P2165R4 v
+        template <typename... _RangeTypes>
+        using _Tuple_or_pair = _STD tuple<_RangeTypes...>;
+#endif // !__cpp_lib_tuple_like
+
+        template <typename _CallbackType>
+        constexpr auto _Create_tuple_transform_lambda(_CallbackType&& _Callback) noexcept {
+            const auto _Created_lambda = [&_Callback]<typename... _ViewTupleTypes>(_ViewTupleTypes&&... _View_tuples) {
+                return _Zip::_Tuple_or_pair<_STD invoke_result_t<_CallbackType&, _ViewTupleTypes>...>{

```

Comment:
MSVC STL generally avoids qualifying `_STD` for types or variable templates.

Reply:
So only use `_STD` for functions, then? I can go ahead and make that change. Does the same format also apply for `_RANGES`? From what I can tell, that seems to be the case, but I would like additional confirmation.

Reply:
Yep, that's correct. Argument-Dependent Lookup specifically affects functions, so non-functions don't need extra qualification, and that applies to `_STD`, `_RANGES`, `_CHRONO` equally.

There are a few places where non-functions are qualified. Most are because we're outside the namespace in question (e.g. in a global-scope `extern "C"` function). A few are avoiding ambiguity with members (e.g. when `reverse_iterator` is a typedef in a class, we say `_STD reverse_iterator` to refer to the actual template, both before and after the typedef is defined in order to avoid any confusion). There are also a few cases where qualification improves throughput (and we've commented them).